### DISCUSSION TITLE
Warning: Current calibration error when setting up a tDCS leadfield

### DISCUSSION THREAD

**Discussion by StevenBm** [simnibs/simnibs#414] [8. Simulation workflows] [answered]
Hi all, 

I was generating a leadfield for a head model in which custom tissues were inserted, specifically conductive burrholes through the skull.  For every simulation, the log tells me that the current calibration error exceeded 10%, with an average of approximately 60%. 

Logically, this error is related to the added holes. However, I am curious how much of an issue this warning is. Could someone help me in estimating the gravity of this warning?

Kind regards,
Steven

**Comment by oulap** [comment]
Hi Steven,

60% calibration error sounds a bit too high. Although the spatial distribution of the e-field is not affected, the magnitude would be off significantly. So that warning definitely needs fixing.

SimNIBS expects the head model to be covered with the scalp tissue (tissue tag 1005). As you mentioned the burr holes are probably the culprit here, especially if the surface tag on the outer surface of the head model changes at the positions of the burr holes. So you would need to relabel everything on the outermost surface to have the tag 1005. 

If you need help to do this, it would be easiest if you could share the head model. If the data is sensitive you can email us at  support@simnibs.org

Best,
Oula

**Reply by StevenBm** [reply]
Dear Oula,

Thank you for your quick reply. The algorithm I use to insert the burrholes approximates the holes by overwriting the skull tissue tags with a custom tissue (1050), so the scalp tissue should be completely intact. I quickly made a version of the Ernie model with two holes in the skull and uploaded it as a zip on Surfdrive, which can be retrieved at https://surfdrive.surf.nl/files/index.php/s/pqsiap23kgN0p6l.

Thank you in advance for your help!

Kind regards,
Steven

**Reply by oulap** [reply]
Sorry for taking so long to reply! I downloaded the mesh and created a leadfield using that. I only got a handful of calibration error warnings (5 or so in total) and the largest calibration error was 12%. I was wondering which conductivity did you assign to the burrhole? I used the conductivity of the CSF (1.654 S/m), which might be a bit too high.

**Reply by StevenBm** [reply]
Dear Oula, 

No problem. Thank you for helping me out! I just checked again on my side what the problem could be and I still had the same issue for the model that I've shared. The conductivities were set to SimNibs standard conductivities and I've used the conductivity of CSF for the burrhole indeed. I've verified that the type of solver does not matter, but the size of the patch does. The errors appear when the size of the patch is 50x50 mm^2 and they drop quite significantly to indeed 12% for a few when the patch size is reduced to 40x40 mm^2. Of course, 50x50 is quite big, but this was set for our clinical studies.

Kind regards,
Steven

**Reply by axelthi** [reply]
Dear Steven,
would you mind sharing a modified ernie model where the errors are high?
Best regards,
Axel

**Reply by StevenBm** [reply]
Dear Axel,

My apologies for my late reaction. In principle, the model that I shared can give high errors when a lead field is created for square electrodes with a size of 50x50 mm^2 (interestingly enough, this error disappears more or less when a size of 40 x 40 mm^2 is used)

Kind regards,
Steven

**Comment by axelthi** [comment, marked as answer]
Dear Steven,

neighboring electrodes will start overlapping on the head when you use electrodes which are too large. All electrodes are added at the beginning before calculating the lead fields. This will cause "overlaps" that can't occur in reality, and this will also make the simulation results invalid.

Best regards,
Axel

**Reply by StevenBm** [reply]
Unfortunately, those were the ones that we had to use for a clinical study. In that case, I presume it's better to optimize with smaller electrodes and use the optimized locations for the bigger electrodes, given that that doesn't cause any overlap between them. Thank you for replying this quickly, I assumed that the lead field was calculated by simulating the field from every possible patch location to a reference separately, which would have explained errors for electrodes closeby to the reference.
